homeassistant/components/tessie/entity.py

```diff
@@ -10,28 +11,72 @@
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN, MODELS
-from .coordinator import TessieStateUpdateCoordinator
+from .coordinator import (
+    TessieEnergySiteInfoCoordinator,
+    TessieEnergySiteLiveCoordinator,
+    TessieStateUpdateCoordinator,
+)
+from .models import TessieEnergyData
 
 
-class TessieEntity(CoordinatorEntity[TessieStateUpdateCoordinator]):
-    """Parent class for Tessie Entities."""
+class TessieBaseEntity(
+    CoordinatorEntity[
+        TessieStateUpdateCoordinator
+        | TessieEnergySiteInfoCoordinator
+        | TessieEnergySiteLiveCoordinator
+    ]
+):
+    """Parent class for Tessie entities."""
 
     _attr_has_entity_name = True
 
     def __init__(
         self,
-        coordinator: TessieStateUpdateCoordinator,
+        coordinator: TessieStateUpdateCoordinator
+        | TessieEnergySiteInfoCoordinator
+        | TessieEnergySiteLiveCoordinator,
         key: str,
     ) -> None:
         """Initialize common aspects of a Tessie entity."""
-        super().__init__(coordinator)
-        self.vin = coordinator.vin
+
         self.key = key
+        self._attr_translation_key = key
+        super().__init__(coordinator)
 
-        car_type = coordinator.data["vehicle_config_car_type"]
+    @property
+    def _value(self) -> Any:
+        """Return value from coordinator data."""
+        return self.coordinator.data.get(self.key)
 
-        self._attr_translation_key = key
+    def get(self, key: str | None = None, default: Any | None = None) -> Any:
+        """Return a specific value from coordinator data."""
+        return self.coordinator.data.get(key or self.key, default)
+
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._async_update_attrs()
+        self.async_write_ha_state()
```

Comment:
```suggestion
    def _handle_coordinator_update(self) -> None:
        """Handle updated data from the coordinator."""
        self._async_update_attrs()
        super()._handle_coordinator_update()
```
Essentially does the same, but I think this is just a better practice :)